qiskit/primitives/backend_estimator_dev.py

```diff
@@ -0,0 +1,723 @@
+        cls,
+        backend: Backend,  # pylint: disable=unused-argument
+        **kwargs,  # pylint: disable=unused-argument
+    ):
+        self = super().__new__(cls)
+        return self
+
+    def _call(
+        self,
+        circuits: Sequence[int],
+        observables: Sequence[int],
+        parameter_values: Sequence[Sequence[float]],
+        **run_options,
+    ) -> EstimatorResult:
+        raise NotImplementedError("This method has been deprecated, use `run` instead.")
+
+
+################################################################################
+## OBSERVABLE DECOMPOSER
+################################################################################
+class _ObservableDecomposer(ABC):
+    """Strategy interface for decomposing observables and getting associated measurement bases."""
+
+    def decompose(self, observable: BaseOperator | PauliSumOp) -> tuple[SparsePauliOp]:
+        """Decomposes a given observable into singly measurable components.
+
+        Note that component decomposition is not unique, for instance, commuting components
+        could be grouped together in different ways (i.e. partitioning the set).
+
+        Args:
+            observable: the observable to decompose into its core components.
+
+        Returns:
+            A list of observables each of which measurable with a single quantum circuit
+            (i.e. on a singlet Pauli basis).
+        """
+        observable = init_observable(observable)
+        return self._decompose(observable)
+
+    @abstractmethod
+    def _decompose(
+        self,
+        observable: SparsePauliOp,
+    ) -> tuple[SparsePauliOp]:
+        ...
+
+    def extract_pauli_basis(self, observable: BaseOperator | PauliSumOp) -> PauliList:
+        """Extract Pauli basis for a given observable.
+
+        Note that the resulting basis may be overcomplete depending on the implementation.
+
+        Args:
+            observable: an operator for which to obtain a Pauli basis for measurement.
+
+        Returns:
+            A `PauliList` of operators serving as a basis for the input observable. Each
+            entry conrresponds one-to-one to the components retrieved from `.decompose()`.
+        """
+        components = self.decompose(observable)
+        paulis = tuple(self._extract_singlet_basis(component) for component in components)
+        return PauliList(paulis)  # TODO: Allow `PauliList` from generator
+
+    @abstractmethod
+    def _extract_singlet_basis(self, observable: SparsePauliOp) -> Pauli:
+        """Extract singlet Pauli basis for a given observable.
+
+        The input observable comes from `._decompose()`, and must be singly measurable.
+        """
+        ...
+
+
+class _NaiveDecomposer(_ObservableDecomposer):
+    """Trivial observable decomposition without grouping components."""
+
+    def _decompose(
+        self,
+        observable: SparsePauliOp,
+    ) -> tuple[SparsePauliOp]:
+        return tuple(observable)
```

Comment:
Main reason for this to be immutable is that the decomposition is fixed, and it does not make sense to mutate it. If you you change it will no longer be the decomposition so better to have it as immutable to avoid accidental updates.